tests/resources/test_material_run.py

```diff
@@ -80,6 +80,8 @@ def test_nomutate_gemd(collection, session):
     # Then
     assert before == after
```

Comment:
I'm actually a little concerned about what `==` means in this context.  Is this all the behavior we actually wanted?

Reply:
I believe that it inherits from [DictSerializable](https://github.com/CitrineInformatics/gemd-python/blob/main/gemd/entity/dict_serializable.py#L138).  Value based equality seems appropriate here.